src/key_info_managers/sqlite_manager/mod.rs

```diff
@@ -70,28 +65,171 @@ impl SQLiteKeyInfoManager {
+        let mut check_for_tables_stmt = conn.prepare(
+            "
+            SELECT
+                *
+            FROM
+                sqlite_master
+            WHERE
+                type='table'
+                AND (
+                    name='key_mapping'
+                    OR name='kim_metadata'
+                )
+        ",
+        )?;
+        let key_iter = check_for_tables_stmt.query_map([], |_row| Ok(()))?;
+        let num_of_tables = key_iter.count();
+        match num_of_tables {
+            // Create tables as they do not exist.
+            0 => {
+                let _ = conn.execute(
+                    "
+                    CREATE TABLE kim_metadata (
+                        id                    TEXT NOT NULL,
+                        int_value             INTEGER NOT NULL,
+                        PRIMARY KEY (id)
+                    )
+                    ",
+                    [],
+                )?;
+                let _ = conn.execute(
+                    "
+                    INSERT INTO
+                        kim_metadata
+                        (id, int_value)
+                    VALUES
+                        ('schema_version', ?1)
+                    ",
+                    params![CURRENT_SCHEMA_VERSION],
+                )?;
+                let _ = conn.execute(
+                    "
+                    CREATE TABLE IF NOT EXISTS key_mapping (
+                        authenticator_id      INTEGER NOT NULL,
+                        application_name      TEXT NOT NULL,
+                        key_name              TEXT NOT NULL,
+                        provider_uuid         TEXT NOT NULL,
+                        provider_name         TEXT NOT NULL,
+                        key_info              BLOB NOT NULL,
+                        key_info_version      INTEGER NOT NULL,
+                        PRIMARY KEY (authenticator_id, application_name, key_name)
+                    )
+                    ",
+                    [],
+                )?;
+            }
+            // The KIM expects both the kim_metadata and key_mapping table to be present, throw an error
+            1 => {
+                let error_message = format!(
+                    "SQLiteKeyInfoManager database schema is not in a recognised format.
+                    A table such as kim_metadata or key_mapping may be missing.
+                    Database found at {}",
+                    database_path
+                        .into_os_string()
+                        .into_string()
+                        .unwrap_or_else(|_| "DB_FILE_PATH_UNKNOWN".to_string()),
+                );
+                error!("{}", error_message);
+                return Err(Error::new(ErrorKind::Other, error_message).into());
+            }
+            // The correct number of tables are present, no-op
+            2 => {}
+            // Shouldn't be possible to reach this. If it does, throw an error
+            _ => {
```

Comment:
could you skip `1` above and just let it fall under this? Seems like the code is identical.